src/brogue/PowerTables.c

```diff
@@ -202,87 +204,17 @@ fixpt defenseFraction(fixpt netDefense) {
-    const short duration[NUMBER_CHARM_KINDS] = {
-        3,  // Health
-        20, // Protection
-        7,  // Haste
-        10, // Fire immunity
-        5,  // Invisibility
-        25, // Telepathy
-        10, // Levitation
-        0,  // Shattering
-        18, // Guardian
-        0,  // Teleportation
-        0,  // Recharging
-        0,  // Negation
-    };
-    const fixpt *increment[NUMBER_CHARM_KINDS] = {
-        POW_0_CHARM_INCREMENT,      // Health
-        POW_0_CHARM_INCREMENT,      // Protection
-        POW_120_CHARM_INCREMENT,    // Haste
-        POW_125_CHARM_INCREMENT,    // Fire immunity
-        POW_120_CHARM_INCREMENT,    // Invisibility
-        POW_125_CHARM_INCREMENT,    // Telepathy
-        POW_125_CHARM_INCREMENT,    // Levitation
-        POW_0_CHARM_INCREMENT,      // Shattering
-        POW_0_CHARM_INCREMENT,      // Guardian
-        POW_0_CHARM_INCREMENT,      // Teleportation
-        POW_0_CHARM_INCREMENT,      // Recharging
-        POW_0_CHARM_INCREMENT,      // Negation
-    };
-
-    short idx = clamp(enchant - 1, 0, LAST_INDEX(POW_125_CHARM_INCREMENT));
-    return duration[charmKind] * increment[charmKind][idx] / FP_FACTOR;
+
+    short idx = clamp(enchant - 1, 0, CHARM_EFFECT_DURATION_INCREMENT_ARRAY_SIZE - 1);
+    return charmEffectTable[charmKind].effectDurationBase * charmEffectTable[charmKind].effectDurationIncrement[idx] / FP_FACTOR;
 }
 
 short charmRechargeDelay(short charmKind, short enchant) {
-    const short duration[NUMBER_CHARM_KINDS] = {
-        2500,   // Health
-        1000,   // Protection
-        800,    // Haste
-        800,    // Fire immunity
-        800,    // Invisibility
-        800,    // Telepathy
-        800,    // Levitation
-        2500,   // Shattering
-        700,    // Guardian
-        920,    // Teleportation
-        10000,  // Recharging
-        2500,   // Negation
-    };
-    const fixpt base[NUMBER_CHARM_KINDS] = {
-        FP_FACTOR * 55 / 100, // Health
-        FP_FACTOR * 60 / 100, // Protection
-        FP_FACTOR * 65 / 100, // Haste
-        FP_FACTOR * 60 / 100, // Fire immunity
-        FP_FACTOR * 65 / 100, // Invisibility
-        FP_FACTOR * 65 / 100, // Telepathy
-        FP_FACTOR * 65 / 100, // Levitation
-        FP_FACTOR * 60 / 100, // Shattering
-        FP_FACTOR * 70 / 100, // Guardian
-        FP_FACTOR * 60 / 100, // Teleportation
-        FP_FACTOR * 55 / 100, // Recharging
-        FP_FACTOR * 60 / 100, // Negation
-    };
 
     enchant = clamp(enchant, 1, 50);
     short delay = charmEffectDuration(charmKind, enchant)
-        + (duration[charmKind] * fp_pow(base[charmKind], enchant) / FP_FACTOR);
-    return max(1, delay);
+        + (charmEffectTable[charmKind].rechargeDelayDuration * fp_pow(charmEffectTable[charmKind].rechargeDelayBase, enchant) / FP_FACTOR);
+    return max(charmEffectTable[charmKind].rechargeDelayMinTurns, delay);
```

Comment:
rechargeDelayMinTurns is a new RapidBrogue concept to floor how quickly powerful charms can recharge (teleportation in particular)